tests/gnrc_tcp/tests/03-send_data.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2018   Simon Brummer <simon.brummer@posteo.de>
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+import os
+import sys
+import socket
+import threading
+import testrunner
+
+from shared_func import generateHostPortNumber
+from shared_func import getHostTapDevice
+from shared_func import getHostLinkLocalAddr
+from shared_func import getRiotInterfaceId
+from shared_func import setupTcpBuffer
+from shared_func import addDataToTcpBuffer
+
+def tcpServer(targetPort, expectedData):
+    sock = socket.socket(socket.AF_INET6, socket.SOCK_STREAM)
+    sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    sock.bind(('::', targetPort))
+    sock.listen(1)
+
+    conn, addr = sock.accept()
+
+    # Receive and verify data
+    recv = conn.recv(len(expectedData), socket.MSG_WAITALL).decode('utf-8')
+
+    if recv != expectedData:
+        raise Exception('Failed to verify data received from RIOT Node.')
+
+    conn.close()
+    sock.close()
```

Comment:
How about having the following function in `shared_func`:

```py
def tcp_server(target_port, test_func=lambda c, *args: None, *test_args):
    with socket.socket(socket.AF_INET6, socket.SOCK_STREAM) as sock:
        sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
        sock.bind(('::', target_port))
        sock.listen(1)

        conn, addr = sock.accept()
        test_func(conn, *test_args)
        conn.close()
```

and then just having the different functionalities implemented in function within your test scripts, which you can provide as a parameter when you start the thread. Same might be interesting for `tcp_client()`.

Reply:
I did it differently now. Wait for next update.

Reply:
:+1: object oriented. Even better! :-)